### PR TITLE
GH-90699: Remove `_Py_IDENTIFIER` usage from `_elementtree` module

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-90699

Automerge-Triggered-By: GH:ericsnowcurrently

### CODE REVIEW

File: Modules/_elementtree.c [hunk 1]
```diff
@@ -84,6 +83,15 @@ typedef struct {
...
     PyObject *pi_factory;
+    /* Interned strings */
+    PyObject *str_text;
+    PyObject *str_tail;
+    PyObject *str_append;
+    PyObject *str_find;
+    PyObject *str_findtext;
+    PyObject *str_findall;
+    PyObject *str_iterfind;
+    PyObject *str_doctype;
```
Comment (ericsnowcurrently) [inline Modules/_elementtree.c:94]: This is a good strategy. 😄

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Comment by kumaraditya303** [comment]
Merging main for https://github.com/python/cpython/pull/98988

**Comment by kumaraditya303** [comment]
Okay so finally CI passed so good to merge.

**Pull request merged**